### ISSUE TITLE
CA1058: False positive

### ISSUE THREAD

**Issue by drieseng** [dotnet/roslyn-analyzers#6991]
### Analyzer

**Diagnostic ID**: [CA1058](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1058): `Avoid dead conditional code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.100-rc.1.23455.8]

### Describe the bug

Compile and execute the following fragment:
```cs
public static class Program
{
    private static void Main()
    {
        Do("-X");
    }

    private static void Do(string value)
    {
        int sign = 1;

        if (string.CompareOrdinal(value, 0, "-", 0, 1) == 0)
        {
            sign = -1;
        }

        if (sign == -1)
        {
            Console.WriteLine("Negative");
        }
    }
}
```

Expected behavior:

* The CA1508 diagnostic is not produced by the compiler.
* The word "Negative" is written to standard output.

Actual behavior:

* The CA1508 diagnostic is produced by the compiler:
```txt
Program.cs(17,13,17,23): warning CA1508: 'sign == -1' is always 'true'. Remove or refactor the condition(s) to avoid dead code.
```
* The word "Negative" is written to standard output.
